app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsContributionsFragment.kt

```diff
@@ -0,0 +1,500 @@
+
+                        if (!qLangMap.containsKey(userContribution.title)) {
+                            qLangMap[userContribution.title] = HashSet()
+                        }
+                        continuedArticlesContributions.add(Contribution(userContribution.title, "", "", EDIT_TYPE_ARTICLE_DESCRIPTION,
+                                "", DateUtil.iso8601DateParse(userContribution.timestamp), WikiSite.forLanguageCode(descLang), 0, 0))
+
+                        qLangMap[userContribution.title]!!.add(descLang)
+                    }
+                    ServiceFactory.get(WikiSite(Service.WIKIDATA_URL)).getWikidataLabelsAndDescriptions(qLangMap.keys.joinToString("|"))
+                            .subscribeOn(Schedulers.io())
+                            .observeOn(AndroidSchedulers.mainThread())
+                }.subscribe({
+                    for (entityKey in it.entities().keys) {
+                        val entity = it.entities()[entityKey]!!
+                        for (contribution in continuedArticlesContributions) {
+                            if (contribution.qNumber == entityKey) {
+                                contribution.title = entity.labels()[contribution.wikiSite.languageCode()]!!.value()
+                                contribution.description = entity.descriptions()[contribution.wikiSite.languageCode()]!!.value()
+                            }
+                        }
+                    }
+                    getArticleContributionDetails()
+                }, { t ->
+                    L.e(t)
+                    showError(t)
+                }))
+    }
+
+    private fun getArticleContributionDetails() {
+        disposables.add(Observable.fromIterable(continuedArticlesContributions)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doFinally {
+                    articleContributions.addAll(continuedArticlesContributions)
+                    if (editFilterType == EDIT_TYPE_ARTICLE_DESCRIPTION) {
+                        createConsolidatedList()
+                    } else {
+                        getImageContributions()
+                    }
+                }
+                .flatMap { contribution -> ServiceFactory.getRest(contribution.wikiSite).getSummary(null, contribution.title) }
+                .subscribe({ summary ->
+                    for (contributionObject in continuedArticlesContributions) {
+                        if (contributionObject.title == summary.displayTitle) {
+                            contributionObject.description = StringUtils.defaultString(summary.description)
+                            contributionObject.imageUrl = summary.thumbnailUrl.toString()
+                            contributionObject.pageViews = pageViewsMap[contributionObject.title]
+                                    ?: 0
+                        }
+                    }
+                }, { t ->
```

Comment:
You may use this:
```
.flatMap({ contribution: Contribution -> ServiceFactory.getRest(contribution.wikiSite).getSummary(null, contribution.title) }, { first: Contribution, second: PageSummary -> Pair(first, second) })
```

The `Pair()` maybe not necessary since you can do the update inside it, and move the logic inside `doFinally` back to the `subscribe。